src/pieces/counting.ts

```diff
@@ -0,0 +1,127 @@
+import { Client, MessageEmbed, TextChannel } from 'discord.js';
+import { DB, CHANNELS } from '@root/config';
+import { Counter } from '../lib/types/Counter';
+
+let countingDB: Counter;
+let countingChannel;
+
+async function register(bot: Client): Promise<void> {
+	countingDB = await bot.mongo.collection(DB.COUNTING).findOne({ piece: 'counting' });
+	countingChannel = bot.channels.cache.get(CHANNELS.COUNTING_CHANNEL) as TextChannel;
+
+	bot.on('messageUpdate', async msg => {
+		if (msg.channel.id !== CHANNELS.COUNTING_CHANNEL) return;
+		if (msg.author.bot && !msg.partial) return; // sage will crash if it hits this line and the message is a partial
+		if (msg.createdTimestamp <= countingDB.startDate) return;
+
+		const endMsg = `${msg.partial ? 'Someone' : msg.author.username} edited a message and broke the count!`;
+
+		if (countingDB.count > 0) {
+			await endGame(endMsg, bot);
+		}
+	});
+
+	bot.on('messageDelete', async msg => {
+		if (msg.channel.id !== CHANNELS.COUNTING_CHANNEL) return;
+		if (msg.author.bot && !msg.partial) return; // sage will crash if it hits this line and the message is a partial
+		if (msg.createdTimestamp <= countingDB.startDate) return;
+
+		const endMsg = `${msg.partial ? 'Someone' : msg.author.username} deleted a message and broke the count!`;
+
+		if (countingDB.count > 0) {
+			await endGame(endMsg, bot);
+		}
+	});
+
+	bot.on('messageCreate', async msg => {
+		if (msg.author.bot) return;
+
+		if (msg.channel.id === CHANNELS.COUNTING_CHANNEL) {
```

Comment:
It's probably a bit cleaner to invert this check and combine it with the one above into one if statement.